biz.aQute.tester.junit-platform/src/aQute/tester/junit/platform/JUnitEclipseListener.java

```diff
@@ -41,6 +40,109 @@
 import org.opentest4j.MultipleFailuresError;
 
 public class JUnitEclipseListener implements TestExecutionListener, Closeable {
+
+	static final Class<? extends AssertionError>	JUNIT3_COMPARISON_FAILURE;
+	static final Class<? extends AssertionError>	JUNIT4_COMPARISON_FAILURE;
+	static final Class<? extends AssertionError>	ASSERTION_FAILED_ERROR;
+	static final Class<? extends AssertionError>	MULTIPLE_FAILURES_ERROR;
+
+	@SuppressWarnings("unchecked")
+	static Class<? extends AssertionError> tryLoad(String name) {
+		try {
+			return (Class<? extends AssertionError>) Class.forName(name);
+		} catch (ClassNotFoundException cnfe) {
+			return null;
+		}
+	}
+
+	static {
+		JUNIT3_COMPARISON_FAILURE = tryLoad("junit.framework.ComparisonFailure");
+		JUNIT4_COMPARISON_FAILURE = tryLoad("org.junit.ComparisonFailure");
+		ASSERTION_FAILED_ERROR = tryLoad("org.opentest4j.AssertionFailedError");
+		MULTIPLE_FAILURES_ERROR = tryLoad("org.opentest4j.MultipleFailuresError");
+	}
+
+	static class ExpectedActualBuilder implements Predicate<Throwable> {
+
+		StringBuilder	expectedBuilder	= new StringBuilder();
+		StringBuilder	actualBuilder	= new StringBuilder();
+		boolean			first			= true;
+
+		void addExpectedActual(String expected, String actual) {
+			if (first) {
+				first = false;
+			} else {
+				expectedBuilder.append("\n\n");
+				actualBuilder.append("\n\n");
+			}
+			expectedBuilder.append(expected);
+			actualBuilder.append(actual);
+		}
+
+		@Override
+		public boolean test(Throwable exception) {
+			// NOTE:
+			// The code in this method is carefully structured to avoid
+			// NoClassDefFoundError. The null test for each if first
+			// confirms that the class is available before proceeding into
+			// the block, so that by that stage it knows that the class
+			// is available and that NCDFE will not be thrown. On the other
+			// hand, if the class is not on the classpath then the code
+			// will never be executed and the thread context classloader
+			// will not attempt to load the class, so again no NCDFE.
+			//
+			// This is particularly important for the JUnit 3/4 comparison
+			// failure assertions, because the OpenTest4J ones will be on the
+			// classpath courtesy of junit-platform-launcher's dependencies.
+			if (ASSERTION_FAILED_ERROR != null && ASSERTION_FAILED_ERROR.isAssignableFrom(exception.getClass())) {
```

Comment:
With the above change to `Predicate<Object>`, the if test becomes more simple and we avoid repeating the test logic:

```java
if (ASSERTION_FAILED_ERROR.test(exception)) {
```